cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_initial.spec.js

```diff
@@ -0,0 +1,261 @@
+          'include',
+          `/app/workspace_create`
+        );
+        cy.location('hash').should('include', 'useCase=Search');
+
+        cy.getElementByTestId('workspaceUseCase-search')
+          .get(`input[type="radio"]`)
+          .should('be.checked');
+      });
+
+      it('should navigate to workspace list page with use case filter', () => {
+        // Click will all button in the use case card
+        cy.getElementByTestId(
+          'workspace-initial-useCaseCard-observability-button-view'
+        ).click();
+        cy.location('pathname', { timeout: 6000 }).should(
+          'include',
+          `/app/workspace_list`
+        );
+        cy.location('hash').should('include', 'useCase=Observability');
+        cy.contains(workspaceName).should('exist');
+
+        // Default filtering based on use cases
+        cy.get('input[type="search"]').should(
+          'have.value',
+          'useCase:"Observability"'
+        );
+
+        miscUtils.visitPage(`/app/home`);
+        // Click will all workspace button
+        cy.contains('View all workspaces').click();
+        cy.location('pathname', { timeout: 6000 }).should(
+          'include',
+          `/app/workspace_list`
+        );
+
+        cy.contains(workspaceName).should('exist');
+        cy.get('input[type="search"]').should('have.value', '');
+      });
+    });
+
+    if (Cypress.env('SECURITY_ENABLED')) {
+      describe('Non OSD admin user visits workspace initial page', () => {
+        const originalUser = ADMIN_AUTH.username;
+        const originalPassword = ADMIN_AUTH.password;
+        before(() => {
+          ADMIN_AUTH.newUser = originalUser;
+          ADMIN_AUTH.newPassword = originalPassword;
+          cy.createInternalUser(
+            NONE_DASHBOARDS_ADMIN_USERNAME,
+            workspaceTestUser
+          );
+          cy.createRole(WORKSPACE_TEST_ROLE_NAME, workspaceTestRole);
+          cy.createRoleMapping(
+            WORKSPACE_TEST_ROLE_NAME,
+            workspaceTestRoleMapping
+          );
+        });
+        beforeEach(() => {
+          // Login as non OSD admin user
+          ADMIN_AUTH.newUser = NONE_DASHBOARDS_ADMIN_USERNAME;
+          ADMIN_AUTH.newPassword = workspaceTestUser.password;
+          // Visit workspace initial page
+          miscUtils.visitPage(`/app/home`);
+        });
+        after(() => {
+          ADMIN_AUTH.newUser = originalUser;
+          ADMIN_AUTH.newPassword = originalPassword;
+          cy.deleteRoleMapping(WORKSPACE_TEST_ROLE_NAME);
+          cy.deleteInternalUser(NONE_DASHBOARDS_ADMIN_USERNAME);
+          cy.deleteRole(WORKSPACE_TEST_ROLE_NAME);
+        });
+
+        it('should show correct no workspaces content', () => {
+          cy.contains(
+            'Request a workspace owner to add you as a collaborator.'
+          ).should('exist');
+        });
+
+        it('should not show create workspace button', () => {
+          cy.getElementByTestId(
```

Comment:
Shall we add an assertion here to see if the page has been loaded first?

Reply:
Sure, the code has been updated